### PR TITLE
Move stencil into each framebuffer

### PR DESCRIPTION
**Pull request by ErikReider**
Grabs some fixes from #144

Also fixes a stencil bug resulting in something like this:

https://github.com/WillPower3309/swayfx/assets/35975961/769aab4c-80fe-428e-8f82-edbf5384f537

### CODE REVIEW

File: sway/desktop/fx_renderer/fx_framebuffer.c [hunk 1]
```diff
@@ -5,7 +5,17 @@ void fx_framebuffer_bind(struct fx_framebuffer *buffer) {
...
+	buffer->fb = -1;
+	buffer->stencil_buffer = -1;
+	buffer->texture.id = 0;
+	buffer->texture.target = 0;
+	buffer->texture.width = -1;
+	buffer->texture.height = -1;
+}
+
+void fx_framebuffer_create(struct fx_framebuffer *buffer, int width, int height,
+		bool bind, bool create_stencil_buffer) {
```
Comment (WillPower3309) [inline sway/desktop/fx_renderer/fx_framebuffer.c:18]: what are your thoughts on removing the bind argument? It'd clean up the creation function a bit, and make it a bit more clear elsewhere in the codebase when we are binding a buffer as we'd need to explicitly call that function after creation.

File: sway/desktop/fx_renderer/fx_framebuffer.c [hunk 2]
```diff
@@ -5,7 +5,17 @@ void fx_framebuffer_bind(struct fx_framebuffer *buffer) {
 	glBindFramebuffer(GL_FRAMEBUFFER, buffer->fb);
 }
 
-void fx_framebuffer_create(struct fx_framebuffer *buffer, int width, int height, bool bind) {
+void fx_framebuffer_init(struct fx_framebuffer *buffer) {
```
Comment (WillPower3309) [inline sway/desktop/fx_renderer/fx_framebuffer.c:8]: Let's have this return a newly created fx_framebuffer, and rename is to fx_framebuffer_create. We can also do any other initial work here. We can then update the existing `fx_framebuffer_create`, to be `fx_framebuffer_update`.

**Review by WillPower3309 — COMMENTED** [review]
Couple comments on structuring fx_framebuffer as a whole. I think storing the stencil in the buffers is the way to go

**Comment by WillPower3309** [comment]
Happy to implement these changes myself this evening, just curious to hear your thoughts

File: sway/desktop/fx_renderer/fx_framebuffer.c (hunk 1 above)
Comment (ErikReider) [reply sway/desktop/fx_renderer/fx_framebuffer.c:18]: I 100% agree, I'm not sure if I like the `create_stencil_buffer` either. Any ideas? Maybe a `fx_framebuffer_create_stencil` instead?

**Comment by ErikReider** [comment]
> Happy to implement these changes myself this evening, just curious to hear your thoughts

Go for it! :)

**Review by WillPower3309 — APPROVED** [review]

**Pull request merged**